CLANG64 doesn't build

### Operating System

Windows 11

### OpenRCT2 build

Latest at time of writing

### Base game

RollerCoaster Tycoon 2

### Area(s) with this issue?

This is a development issue

### Describe the issue

CLANG64 fails to build

### Steps to reproduce

mkdir build & cd build
cmake ..
ninja

### Attachments

```
FAILED: [code=1] CMakeFiles/libopenrct2.dir/src/openrct2/core/FileWatcher.cpp.obj
C:/msys64/clang64/bin/ccache.exe C:\msys64\clang64\bin\c++.exe -DDOCDIR="\"C:/Program Files (x86)/openrct2/share/doc/openrct2\"" -DENABLE_SCRIPTING -IC:/msys64/home/Cal/OpenRCT2/libopenrct2 -isystem C:/msys64/home/Cal/OpenRCT2/src/openrct2/../thirdparty/quickjs-ng -isystem C:/msys64/clang64/include/freetype2 -isystem C:/msys64/clang64/include/libpng16 -isystem C:/msys64/clang64/include/harfbuzz -isystem C:/msys64/clang64/include/glib-2.0 -isystem C:/msys64/clang64/lib/glib-2.0/include -isystem C:/msys64/home/Cal/OpenRCT2/src/openrct2/../thirdparty -fstrict-overflow -fstrict-aliasing -Werror -Wundef -Wmissing-declarations -Winit-self -Wall -Wextra -Wshadow -Wno-unknown-pragmas -Wno-missing-braces -Wno-comment -Wnonnull -Wno-unused-parameter -Wno-attributes -DDEBUG=0 -fno-char8_t -Wno-deprecated-declarations -Wmissing-prototypes -D__USE_MINGW_ANSI_STDIO=1 -O3 -DNDEBUG -std=gnu++20   -fstack-protector-strong -Wsuggest-override -Wnon-virtual-dtor -Wmissing-variable-declarations -Wmissing-field-initializers -Wunreachable-code-break -Wrange-loop-analysis -Wtautological-unsigned-zero-compare -Wold-style-cast -Wignored-qualifiers -Wstrict-overflow=1 -pthread -MD -MT CMakeFiles/libopenrct2.dir/src/openrct2/core/FileWatcher.cpp.obj -MF CMakeFiles\libopenrct2.dir\src\openrct2\core\FileWatcher.cpp.obj.d -o CMakeFiles/libopenrct2.dir/src/openrct2/core/FileWatcher.cpp.obj -c C:/msys64/home/Cal/OpenRCT2/src/openrct2/core/FileWatcher.cpp
In file included from C:/msys64/home/Cal/OpenRCT2/src/openrct2/core/FileWatcher.cpp:36:
C:/msys64/home/Cal/OpenRCT2/src/openrct2/core/FileWatcher.h:72:10: error: private field '_finished' is not used [-Werror,-Wunused-private-field]
   72 |     bool _finished{};
      |          ^
1 error generated.
```

Copying over the data folder from the github release makes it work fine, but I would like to find out how to actually build a functioning copy.
Running "DESTDIR=. ninja install" from the build instructions for CLANG64 causes this error.
```
CMake Error at C:/msys64/home/user/OpenRCT2/cmake/download.cmake:11 (file):
  file failed to create directory:

    C:/msys64/home/user/OpenRCT2/build/.C:/Program

  because: Invalid argument
Call Stack (most recent call first):
  cmake_install.cmake:43 (download_openrct2_zip)



FAILED: [code=1] CMakeFiles/install.util
C:\WINDOWS\system32\cmd.exe /C "cd /D C:\msys64\home\user\OpenRCT2\build && C:\msys64\clang64\bin\cmake.exe -P cmake_install.cmake"
ninja: build stopped: subcommand failed.
```

--- a/src/openrct2/core/FileWatcher.h
+++ b/src/openrct2/core/FileWatcher.h
@@ -68,7 +68,7 @@ public:
     ~FileWatcher();
 
 private:
-#if defined(_WIN32) || defined(__linux__)
+#if defined(__linux__)
     bool _finished{};
 #elif defined(__APPLE__)
     static void FSEventsCallback(

